### PR TITLE
11 games

### PR DESCRIPTION
**Pull request by 333Abe**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -75,10 +75,68 @@ describe("GET api/reviews", () => {
...
   });
+  test("200: accepts category query and returns only reviews with a matching category", () => {
+    return request(app)
+      .get("/api/reviews?category=dexterity")
+      .expect(200)
+      .then(({ body: { reviews } }) => {
+        expect(reviews).toHaveLength(1);
+      });
+  });
+  test("200: accepts sort_by and order queries and returns ordered results", () => {
```
Comment (popatre) [inline __tests__/app.test.js:86]: As the test has an `and` in it, normally tells you it should be two tests

File: __tests__/app.test.js [hunk 2]
```diff
@@ -75,10 +75,68 @@ describe("GET api/reviews", () => {
...
+      .get("/api/reviews?sort_by=category&order=asc")
+      .expect(200)
+      .then(({ body: { reviews } }) => {
+        expect(reviews).toHaveLength(13);
+        expect(reviews).toBeSortedBy("category");
+      });
+  });
+  test("200: accepts category query and returns only reviews with a matching category, sorted by defined field (default DESC)", () => {
+    return request(app)
+      .get("/api/reviews?category=social deduction&sort_by=title")
```
Comment (popatre) [inline __tests__/app.test.js:97]: im surprised this works with a space

File: __tests__/app.test.js [hunk 3]
```diff
@@ -75,10 +75,68 @@ describe("GET api/reviews", () => {
...
+        expect(body.msg).toBe("Bad request");
+      });
+  });
+  test("404 non-existent category", () => {
+    return request(app)
+      .get("/api/reviews?category=invalid")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Not found");
+      });
```
Comment (popatre) [inline __tests__/app.test.js:134]: Looks good

File: controllers/reviews.controllers.games.js [hunk 4]
```diff
@@ -8,10 +8,39 @@ const {
...
+    "created_at",
+    "votes",
+  ];
+  const validOrders = ["DESC", "ASC"];
+  if (order !== undefined) order = order.toUpperCase();
+  if (order === undefined) order = "DESC";
+  if (sort_by === undefined) sort_by = "created_at";
+
+  if (!validOrders.includes(order) || !validSortBys.includes(sort_by)) {
+    res.status(400).send({ msg: "Bad request" });
```
Comment (popatre) [inline controllers/reviews.controllers.games.js:31]: All this data handling stuff would be done in the model. You;d then reject and let the error handler deal with it

**Review by popatre — APPROVED** [review]
Approved. But theres a lot of data stuff you should ideally be doing in the model

File: __tests__/app.test.js (hunk 2 above)
Comment (333Abe) [reply __tests__/app.test.js:97]: yeah, same

**Pull request merged**